### PR TITLE
Support sending a security message

### PR DESCRIPTION
**Pull request by kfbehar**
## Checklist
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [x] This pull-request is submitted against the `master` branch.

### CODE REVIEW

File: e2e/test/MessageSendE2ETests.cs [hunk 1]
```diff
@@ -36,30 +42,84 @@ public async Task Message_DeviceSendSingleMessage_Amqp()
             await SendSingleMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
         }
 
+        [TestMethod]
+        public async Task SecurityMessage_DeviceSendMessage_Amqp()
```
Comment (abhipsaMisra) [inline e2e/test/MessageSendE2ETests.cs]: Test name format: <Message>_DeviceSendSingleMessage_<transport_type>

File: e2e/test/MessageSendE2ETests.cs [hunk 2]
```diff
@@ -36,30 +42,84 @@ public async Task Message_DeviceSendSingleMessage_Amqp()
...
+        {
+            await TestSecurityMessage(TestDeviceType.Sasl, Client.TransportType.Amqp_Tcp_Only).ConfigureAwait(false);
+        }
+
+        [TestMethod]
+        public async Task SecurityMessage_ModuleSendMessage_Amqp()
```
Comment (abhipsaMisra) [inline e2e/test/MessageSendE2ETests.cs]: Test name format: "Message_ModuleSendSingleMessage_transport_type"

**Comment by timtay-microsoft** [comment]
@kfbehar some of the e2e test cases are failing. SecurityMessage_DeviceSendSingleMessage_AmqpWs and all the other protocol variations on this test 

Microsoft.ServiceBus.Messaging.MessagingCommunicationException: An error occurred during communication with '3548fe0c9139482988d9665c382020e7_B14'. Check the connection information, then retry. ---> System.TimeoutException: The operation did not complete within the allocated time 00:01:10 for object session47. For more information on exception types and proper exception handling, please refer to http://go.microsoft.com/fwlink/?LinkId=761101

**Comment by timtay-microsoft** [comment]
The NetMF build also failed with the below two errors:

         c:\agent\_work\13\s\iothub\device\src\Message.cs(398,138): error CS0103: The name 'StringComparison' does not exist in the current context [C:\agent\_work\13\s\iothub\device\src.NetMF\HttpClient_44\Microsoft.Azure.Devices.Client.Http.NetMF_44.csproj]

         C:\agent\_work\13\s\iothub\device\src\Message.cs(398,138): error CS0103: The name 'StringComparison' does not exist in the current context [C:\agent\_work\13\s\iothub\device\src.NetMF\HttpClient_43\Microsoft.Azure.Devices.Client.Http.NetMF_43.csproj]

**Comment by kfbehar** [comment]
Hi, can you give me access to see the e2e run results? I get a 401.

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

**Pull request merged**